src/dotnet-new3/Program.cs

```diff
@@ -72,7 +74,29 @@ private static DefaultTemplateEngineHost CreateHost(bool emitTimings)
                 typeof(NupkgUpdater).GetTypeInfo().Assembly                         // for assembly: Microsoft.TemplateSearch.Common
             });
 
-            DefaultTemplateEngineHost host = new DefaultTemplateEngineHost(HostIdentifier, HostVersion, CultureInfo.CurrentCulture.Name, preferences, builtIns, new[] { "dotnetcli" });
+            ILogger logger = null;
+            if (enableLogging)
```

Comment:
I think we should always create `logger` and use flags only to control what level of logging it should print.